File: problems/water_flow/submissions/accepted/water_flow.c
#include <stdio.h>

int main(int argc, char const *argv[])
{

    float flow;
    int i, n_valves;

    scanf("%f", &flow);
    scanf("%d", &n_valves);

    int valves_bandwidth[n_valves];
    int delays[n_valves];

    for (i = 0; i < n_valves; i++)
        scanf("%d", &valves_bandwidth[i]);

    float results[n_valves];
    delays[0] = 0;
    results[0] = (flow / valves_bandwidth[0]) + delays[0];

    for (i = 1; i < n_valves; i++)
    {
        scanf("%d", &delays[i]);
        delays[i] += delays[i - 1];
        results[i] = (flow / valves_bandwidth[i]) + delays[i];
    }

    int the_valve;
    //finding the index of min
    int index = 0;
    int min = results[0];
    for (i = 1; i < n_valves; i++)
        if (results[i] < min)
        {
            min = results[i];
            index = i;
        }
    printf("%d", index + 1);
    return 0;
}

File: problems/water_flow/submissions/accepted/Water_flow.java
import java.lang.ProcessBuilder.Redirect.Type;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.Scanner;

public class Water_flow{
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        float flow = Float.parseFloat(sc.nextLine());
        int nValves = Integer.parseInt(sc.nextLine());
        // valve bandiwth
        String[] vBande = sc.nextLine().split("\\s+");
        int[] valvesBandwidth = new int[vBande.length];
        for (int i=0 ; i< vBande.length;i++) {
            valvesBandwidth[i] = Integer.parseInt(vBande[i]);            
        }
        // delays
        String[] del = sc.nextLine().trim().split("\\s+");
        int [] delays = new int[del.length + 1];
        delays[0]= 0;

        for(int i = 0; i< del.length;i++){
            delays[i+1] = Integer.parseInt(del[i]);
        }
        sc.close();
        // solution
        ArrayList<Float> result = new ArrayList<Float>();
        result.add(flow/valvesBandwidth[0]);
        
        for (int i = 1; i< delays.length; i++){
            delays[i] += delays[i-1];
            result.add(flow/valvesBandwidth[i]+ delays[i]);
        }
        int theValve = result.indexOf(Collections.min(result)) + 1;

        System.out.println(theValve);
    }
}